service/controller/app/v1/resource/appcatalog/delete.go

```diff
@@ -0,0 +1,24 @@
+package appcatalog
+
+import (
+	"context"
+
+	"github.com/giantswarm/microerror"
+
+	"github.com/giantswarm/app-operator/service/controller/app/v1/key"
+)
+
+// EnsureDeleted gets the appCatalog CR specified in the provided app CR.
```

Comment:
I don't think we need appCatalog in deletion event, but some resources refer it inside `desired.go`. @rossf7 found out we need to pass it through :)